Source/Result.swift

```diff
@@ -100,3 +100,82 @@ extension Result: CustomDebugStringConvertible {
         }
     }
 }
+
+// MARK: - Functional Tools
+
+extension Result {
+    /// Returns the success value, or throws the failure error.
+    ///
+    ///     let possibleString: Result<String> = .success("success")
+    ///     try print(possibleString.unwrap())
+    ///     // Prints "success"
+    ///
+    ///     let noString: Result<String> = .failure(error)
+    ///     try print(noString.unwrap())
+    ///     // Throws error
+    func unwrap() throws -> Value {
```

Comment:
This is usually known as `dematerialize` and I think we should keep that to maintain consistency with other `Result` types (whether from Swift or other languages). At the same time, a global function called `materialize` that takes a throwing function and wraps it in a `Result` provides the opposite functionality. It would be good to have that too I think.